game/scripts/vscripts/units/swiper.lua

```diff
@@ -0,0 +1,213 @@
+require('abilities/swiper/boss_swiper_swipe')
+
+function Spawn( entityKeyValues )
+	if thisEntity == nil then
+		return
+	end
+
+	if IsServer() == false then
+		return
+	end
+
+	thisEntity.hThrustAbility = thisEntity:FindAbilityByName( "boss_swiper_thrust" )
+	thisEntity.hFrontswipeAbility = thisEntity:FindAbilityByName( "boss_swiper_frontswipe" )
+	thisEntity.hBackswipeAbility = thisEntity:FindAbilityByName( "boss_swiper_backswipe" )
+	thisEntity.hReapersRushAbility = thisEntity:FindAbilityByName( "boss_swiper_reapers_rush" )
+
+	thisEntity:SetContextThink( "SwiperBossThink", SwiperBossThink, 1 )
+end
+
+function SwiperBossThink()
+	if GameRules:IsGamePaused() == true or GameRules:State_Get() == DOTA_GAMERULES_STATE_POST_GAME or thisEntity:IsAlive() == false then
+		return 1
+	end
+
+	-- if not thisEntity:IsIdle() then
+	-- 	return 1.0
+	-- end
+
+	if not thisEntity.bInitialized then
+		thisEntity.vInitialSpawnPos = thisEntity:GetOrigin()
+		thisEntity.bInitialized = true
+	end
+
+	local enemies = FindUnitsInRadius(
+		thisEntity:GetTeamNumber(),
+		thisEntity:GetOrigin(), nil,
+		1000,
+		DOTA_UNIT_TARGET_TEAM_ENEMY,
+		DOTA_UNIT_TARGET_HERO,
+		DOTA_UNIT_TARGET_FLAG_MAGIC_IMMUNE_ENEMIES + DOTA_UNIT_TARGET_FLAG_FOW_VISIBLE,
+		FIND_CLOSEST,
+		false
+	)
+
+	if #enemies == 0 then
+		return RetreatHome()
+	end
+
```

Comment:
this needs code for aggro/deagro/leash

Reply:
- this finds enemies in a 1000 radius of that boss, in that case if you can't run 1000 units from the boss he will follow you forever.
 - after a max distance (BOSS_LEASH_SIZE) from the pit he should return to its position, regardless of having enemies close.
 - this AI will aggro on sight, if you pass 1000 near the pit.

It is not the best example in the world but you can look at ai_lycan_boss or any other boss to see how it is implemented

Reply:
thx will do it like that